### DISCUSSION TITLE
Shiika progress report Mar. 2024

### DISCUSSION THREAD

**Discussion by yhara** [shiika-lang/shiika#540] [Show and tell]
I'm not currently making commits for this repository, but rest assured, I haven't forgotten about Shiika, of course! Lately, I've been delving into concurrency matters (#532).

My initial experiment involves integrating LLVM and Tokio, which I've successfully accomplished in https://github.com/yhara/chiika. Following that, I'm exploring MLIR in another repository, https://github.com/yhara/milika. It took some time to familiarize myself with the MLIR ecosystem, but it's satisfying not to have to manually write branching instructions just to implement 'if' or 'while' statements.

Once I've completed all the experiments I want to conduct in Milika, I'll backport the results to Shiika. I don't have a concrete plan yet for how to do this, but at the very least, lib/skc_parser will remain unchanged. The main part that will be affected is lib/skc_ast2hir. Additionally, lib/skc_codegen may undergo a complete rewrite if I opt for MLIR (although I'm still unsure if it's the best approach).

Stay tuned!

**Comment by yamakoud** [comment]
Nice!!!
